app/src/main/scala/org/alephium/explorer/persistence/queries/QuerySplitter.scala

```diff
@@ -0,0 +1,142 @@
+    *
+    * @param rows                All query parameters
+    * @param initialQuery        Returned when params are empty
+    * @param columnsPerRow       Total number of columns/parameters in a single row of a query.
+    *                            Or the number of '?' in a single row.
+    * @param maxRowsPerQuery     Maximum number of rows in each query. Eg: [[defaultMaxRowsPerQuery]]
+    * @param foldLeft            Given a set of following inputs (Tuple3) returns the next query.
+    *                            Similar to foldLeft in a collection type.
+    *                              - _1 = Parameter split for current query
+    *                              - _2 = Placeholder for current query
+    *                              - _3 = Previous query. Used to join with next query.
+    * @tparam R type of rows
+    * @tparam Q type of query
+    * @return A single batched query
+    */
+  def splitFoldLeft[R, Q](rows: Iterable[R],
+                          initialQuery: Q,
+                          columnsPerRow: Short,
+                          maxRowsPerQuery: Short)(foldLeft: (Iterable[R], String, Q) => Q): Q = {
+    //calculate maximum number of parameters to allow per query
+    val maxParamsPerQuery =
+      maxRowsPerQuery * columnsPerRow
+
+    //maxParamsPerQuery should not exceed max limit allowed by Postgres i.e. postgresDefaultParameterLimit
+    val maxParamsPerQueryAllowed =
+      (maxParamsPerQuery min postgresDefaultParameterLimit.toInt).toShort
+
+    splitFoldLeftLimitByMaxParams[R, Q](
+      rows              = rows,
+      initialQuery      = initialQuery,
+      columnsPerRow     = columnsPerRow,
+      maxParamsPerQuery = maxParamsPerQueryAllowed
+    )(foldLeft)
+  }
+
+  /**
+    * Splits queries into batches limited by maximum/total number of parameters allowed per query.
+    *
+    * @param rows                All query parameters
+    * @param initialQuery        Returned when params are empty
+    * @param columnsPerRow       Total number of columns/parameters in a single row of a query.
+    *                            Or the number of '?' in a single row.
+    * @param maxParamsPerQuery   Maximum number of parameters for each query. Eg: [[postgresDefaultParameterLimit]]
+    * @param foldLeft            Given a set of following inputs (Tuple3) returns the next query.
+    *                            Similar to foldLeft in a collection type.
+    *                              - _1 = Parameter split for current query
+    *                              - _2 = Placeholder for current query
+    *                              - _3 = Previous query. Used to join with next query.
+    * @tparam R type of rows
+    * @tparam Q type of query
+    * @return A single batched query
+    */
+  def splitFoldLeftLimitByMaxParams[R, Q](
+      rows: Iterable[R],
+      initialQuery: Q,
+      columnsPerRow: Short,
+      maxParamsPerQuery: Short)(foldLeft: (Iterable[R], String, Q) => Q): Q = {
```

Comment:
If we pass by `maxRowsPerQuery` instead of `maxParamsPerQuery`, then we will not recalculate `maxRows` for this function.

We could use `val maxRowsPerQueryAllowed =
      maxRowsPerQuery min postgresDefaultParameterLimit.toInt / columnsPerRow`